src/java.base/share/native/libjli/java.c

```diff
@@ -1395,11 +1395,17 @@ ParseArguments(int *pargc, char ***pargv,
     return JNI_TRUE;
 }
 
+/*
+ * Sets the relevant environment variables that are subsequently used by
+ * the ShowSplashScreen() function. The splash_file_path and jar_path parameters
+ * are used to determine which environment variables to set.
+ * The splash_file_path is the value that was provided to the "-splash:" option
+ * when launching java. It may be null, which implies the "-splash:" option wasn't used.
+ * The jar_path is the value that was provided to the "-jar" option when launching java.
+ * It too may be null, which implies the "-jar" option wasn't used.
```

Comment:
But presumably we only call this function if (at least?) one of them is not null?

Reply:
Hello David, the implementation of this function takes the responsibility of acting as a no-op if both the parameters are null. The call site doesn't do that check. Do you prefer the call site to do that check and we update this function's documentation that at least one of them should be non null?

Reply:
No, no, that is fine as-is. I was mis-remembering how the code was structured.